addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -673,10 +673,64 @@ export function getTraversedNodes(editable, range = getDeepRange(editable)) {
     do {
         node = iterator.nextNode();
     } while (node && node !== range.startContainer);
+    if (
+        node &&
+        node.nodeType === Node.ELEMENT_NODE &&
+        node.childNodes &&
+        range.startOffset &&
+        node.childNodes[range.startOffset - 1].nodeName === "BR"
+    ) {
+        // handle the cases:
+        // <p>ab<br>[</p><p>cd</p>] => [p2, cd]
+        // <p>ab<br>[<br>cd</p><p>ef</p>] => [br2, cd, p2, ef]
+        const targetBr = node.childNodes[range.startOffset - 1];
+        while (node != targetBr) {
+            node = iterator.nextNode();
+        }
+        node = iterator.nextNode();
+    }
+    if (
+        node &&
+        !range.collapsed &&
+        node === range.startContainer &&
+        range.startOffset === nodeSize(node) &&
+        node.nextSibling &&
+        node.nextSibling.nodeName === "BR"
+    ) {
+        // handle the case: <p>ab[<br>cd</p><p>ef</p>] => [br, cd, p2, ef]
+        node = iterator.nextNode();
+    }
     const traversedNodes = new Set([node, ...descendants(node)]);
     while (node && node !== range.endContainer) {
         node = iterator.nextNode();
-        node && traversedNodes.add(node);
+        if (node) {
+            if (
+                node !== range.endContainer ||
+                // handle the case: [<p>ab</p><p>cd<br>]ef</p> => [ab, p2, cd, br]
+                !(
+                    range.endOffset === 0 &&
+                    !range.collapsed &&
+                    node.previousSibling &&
+                    node.previousSibling.nodeName === "BR"
+                )
+            ) {
+                traversedNodes.add(node);
+            }
+        }
+    }
+    if (node) {
+        // handle the cases:
+        // [<p>ab</p><p>cd<br>]</p> => [ab, p2, cd, br]
+        // [<p>ab</p><p>cd<br>]<br>ef</p> => [ab, p2, cd, br1]
+        for (const descendant of descendants(node)) {
+            if (
+                descendant.parentElement === node &&
+                childNodeIndex(descendant) >= range.endOffset
+            ) {
+                break;
+            }
+            traversedNodes.add(descendant);
```

Comment:
This is pretty weird to me because it does not change the value of `node` so aren't we adding the same node twice? And actually I have the same feeling with line 703. Aren't we looping on the same nodes afterwards, adding them twice? (which would amount to three total with the suggested change) I know it's a Set so it doesn't matter, and you didn't write those lines but... am I missing something or is it really that unoptimized to begin with? I mean, we could remove the line that adds the descendants of `node` and it would change nothing to the function except making it faster, right?